### PR TITLE
Fix flake `F403` and `F405` errors: don't use star imports

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary
Remove `F403` and `F405` from the list of ignored flake rules in `.flake8`. Replace star imports for explicit imports.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/basic/practices.html#style).
* [ ] Added [tests](https://docs.simpeg.xyz/content/basic/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/basic/practices.html#documentation).
* [ ] Added relevant method tags (i.e. `GRAV`, `EM`, etc.)
* [ ] Marked as ready for review (ff this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Additional information

Would be nice to merge this after #1238. The `F403` and `F405` errors should be removed from the `.flake8` file.

#### TODO

- [x] Update branch after merging #1238
- [x] Remove `F403` from `.flake8`
- [x] Remove `F405` from `.flake8`

### CODE REVIEW

File: out [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DCAnalytic_Pole_Dipole
```
Comment (jcapriot) [inline out:1]: Leftover from a `grep > out` ?

Comment (santisoler) [reply out:1]: 🤦🏼

File: SimPEG/electromagnetics/analytics/__init__.py [hunk 2]
```diff
@@ -1,6 +1,36 @@
 from .TDEM import hzAnalyticDipoleT, hzAnalyticCentLoopT
 from .FDEM import hzAnalyticDipoleF
-from .FDEMcasing import *
-from .DC import *
-from .FDEMDipolarfields import *
+from .FDEMcasing import (
+    getKc,
+    _r2,
```
Comment (jcapriot) [inline SimPEG/electromagnetics/analytics/__init__.py:5]: I know they get imported with the * command, but it's probably unnecessary to import the hidden methods here.

Comment (santisoler) [reply SimPEG/electromagnetics/analytics/__init__.py:5]: I agree. I'll remove them and check if something fails.

**Pull request merged**